inst/templates/ad_advs.R

```diff
@@ -0,0 +1,377 @@
+advs_wgt_age <- derive_params_growth_age(
+  advs,
+  sex = SEX,
+  age = AAGECUR,
+  age_unit = AAGECURU,
+  meta_criteria = weight_for_age,
+  parameter = VSTESTCD == "WEIGHT",
+  set_values_to_sds = exprs(
+    PARAMCD = "WTASDS",
+    PARAM = "Weight-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "WTAPCTL",
+    PARAM = "Weight-for-age percentile"
+  )
+) %>%
+  mutate(
+    PARAMN = case_when(
+      PARAMCD == "WTASDS" ~ 5,
+      PARAMCD == "WTAPCTL" ~ 6,
+      TRUE ~ PARAMN
+    )
+  ) %>%
+  filter(PARAMCD %in% c("WTASDS", "WTAPCTL"))
+
+## Calculate BMI for AGE z-score and Percentile ----
+## Note: PARAMN needs to be updated for z-score and percentile in final dataset.
+advs_bmi_age <- derive_params_growth_age(
+  advs,
+  sex = SEX,
+  age = AAGECUR,
+  age_unit = AAGECURU,
+  meta_criteria = bmi_for_age,
+  parameter = VSTESTCD == "BMI",
+  set_values_to_sds = exprs(
+    PARAMCD = "BMISDS",
+    PARAM = "BMI-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "BMIPCTL",
+    PARAM = "BMI-for-age percentile"
+  )
+) %>%
+  mutate(
+    PARAMN = case_when(
+      PARAMCD == "BMISDS" ~ 7,
+      PARAMCD == "BMIPCTL" ~ 8,
+      TRUE ~ PARAMN
+    )
+  ) %>%
+  filter(PARAMCD %in% c("BMISDS", "BMIPCTL"))
+
+## Calculate Head Circumference for AGE z-score and Percentile ----
+## Note: PARAMN needs to be updated for z-score and percentile in final dataset.
+advs_hdc_age <- derive_params_growth_age(
+  advs,
+  sex = SEX,
+  age = AAGECUR,
+  age_unit = AAGECURU,
+  meta_criteria = who_hc_for_age,
+  parameter = VSTESTCD == "HDCIRC",
+  set_values_to_sds = exprs(
+    PARAMCD = "HDCSDS",
+    PARAM = "HDC-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "HDCPCTL",
+    PARAM = "HDC-for-age percentile"
+  )
+) %>%
+  mutate(
+    PARAMN = case_when(
+      PARAMCD == "HDCSDS" ~ 9,
+      PARAMCD == "HDCPCTL" ~ 10,
+      TRUE ~ PARAMN
+    )
+  ) %>%
+  filter(PARAMCD %in% c("HDCSDS", "HDCPCTL"))
+
+## Calculate for Height/Length ----
+## derive_params_growth_height function not yet ready?
+## It needs to be updated once the function is ready to use.
+## from issue #34
+# advs_wgt_height <- derive_params_growth_height(
+#   advs,
+#   sex = SEX,
+#   age = AAGECUR,
+#   age_unit = AAGECURU,
+#   meta_criteria = weight_for_age,
+#   parameter = VSTESTCD == "HEIGHT",
+#   measure = HEIGHT,
+#   height_age = 730,
+#   set_values_to_sds = exprs(
+#     PARAMCD = "WGTHSDS",
+#     PARAM = "Weight-for-length/height Z-Score"
+#   ),
+#   set_values_to_pctl = exprs(
+#     PARAMCD = "WGTHPCTL",
+#     PARAM = "Weight-for-length/height Percentile"
+#   )
+# )  %>%
+# filter(PARAMCD %in% c("WGTHSDS", "WGTHPCTL"))
+
+## Combine all derived parameters together
+advs <- advs %>%
+  rbind(advs_wgt_age, advs_bmi_age, advs_hdc_age)
+# z-score and percentile for HEIGHT and LENGTH to be added once the
+# `derive_params_growth_height` function is ready
+
+## Get ASEQ ----
```

Comment:
Also derive ANL01FL?

Reply:
How can we know which records are post-baseline without Treatment start date ? Currently no Pediatric ADSL available.

Reply:
ok, we can add later, thanks! i was just comparing vs the admiral ADVS template to see if we missed anything